src/scripts/main.js

```diff
@@ -1,3 +1,51 @@
+/* eslint-disable max-len */
 'use strict';
 
-// write code here
+const list = document.querySelector('ul');
+
+const helperFn = {
+  sort(prev, next) {
+    return helperFn.convertToNumber(next.dataset.salary) - helperFn.convertToNumber(prev.dataset.salary);
+  },
+  convertToNumber(number) {
+    return +number.replace('$', '').replace(',', '.');
+  },
+  appendToList(item) {
+    list.innerHTML += `
+      <li
+        data-position="${item.dataset.position}"
+        data-salary="${item.dataset.salary}"
+        data-age="${item.dataset.age}"
+      >
+      ${item.innerText}
+      </li>
+    `;
+  },
+};
+
+function sortList(array) {
+  const sortedList = [...array.querySelectorAll('li')].sort(helperFn.sort);
+
+  // clear list
+  list.innerHTML = '';
+  // add items
+  sortedList.forEach(item => helperFn.appendToList(item));
```

Comment:
If you just append those sorted nodes to the ul `list.innerHTML = '';` will be redundant
```js
sortedNodes.forEach(node => ul.append(node));
```